public/app/features/profile/SignupCtrl.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { PureComponent } from 'react';
+
+interface SignupProps {
+  children(arg0: ChildProps): JSX.Element;
+}
+
+interface ChildProps {
+  onSubmit(e: React.FocusEvent): void;
+}
+
+export class SignupCtrl extends PureComponent<SignupProps> {
+  onSubmit = () => {};
+
+  render() {
+    const { children } = this.props;
+    return (
+      <>
+        {children({
+          onSubmit: this.onSubmit,
+        })}
+      </>
+    );
+  }
+}
```

Comment:
what is the purpose of this component?

Reply:
It is supposed to handle the backend calls, Redux connection etc. I thought I'd do a similar approach as to the login page.